File: patch.py
#!/usr/bin/env python
# -*- coding: latin-1 -*-
'''PATCH an object on an ENCODE server'''

import sys, requests, json

# Indicate that the content sent to the server is JSON
# Force return from the server in JSON format
HEADERS = {'content-type': 'application/json', 'accept': 'application/json'}

# Authentication is always required to PATCH ENCODE objects
AUTHID = "ABCDEFG" #<- Replace this with your keypair
AUTHPW = "myspecialsecret" #<- Replace this with your keypair

# This URL locates the ENCODE experiment with accession number ENCSR000AJT
URL = "https://test.encodedcc.org/experiments/ENCSR000AJT/"

# Create the JSON to send to the server
payload_dict = {
	"description": "Originally Caltech ChIP-Seq mouse C2C12 EqS_2.0pct_60hr Control_36bp" #Originally Caltech ChIP-Seq mouse C2C12 EqS_2.0pct_60hr Control_36bp
}
json_payload = json.dumps(payload_dict)

# Send the request to the server
response = requests.patch(URL, auth=(AUTHID, AUTHPW), headers=HEADERS, data=json_payload)

# Process the response
if not response.status_code == 200:
	print >> sys.stderr, response.text

# Print the JSON response
print json.dumps(response.json(), indent=4, separators=(',', ': '))

File: patch.php
<?php
# PATCH an object on an ENCODE server'

#Uses Requests library from https://github.com/rmccue/Requests
include('../Requests/library/Requests.php');
Requests::register_autoloader();

# Force return from the server in JSON format
$headers = array('Content-Type' => 'application/json', 'Accept' => 'application/json');

# Authentication is always required to POST ENCODE objects
$authid = "SU45FB2Q"; // <-Replace this with your access_key
$authpw = "rae76sr5bntlz5c6"; // <-Replace this with your secret_access_key
$auth = array('auth' => array($authid, $authpw));

# This URL locates the ENCODE experiment that user can submit too 
$url ="<server>/experiments/<experiment accession/" ; // <-Replace this with appropriate server and experiment accession

# Build a array with the experiment metadata
$path_experiment= array(
	"description" => "PATCH example experiment"
);

#POST the JSON and get back response
$response = Requests::patch($url, $headers, json_encode($path_experiment), $auth);

# If the POST succeeds, the response is the new object in JSON format
var_dump($response->body);

?>